sdk/tables/azure-tables/src/generated/models/index.ts

```diff
@@ -45,23 +45,51 @@ export interface TableResponseProperties {
 }
 
 /**
- * The properties for creating a table.
+ * Table Service error.
  */
-export interface TableProperties {
+export interface TableServiceError {
   /**
-   * The name of the table to create.
+   * The odata error.
    */
-  tableName?: string;
+  odataError?: TableServiceErrorOdataError;
```

Comment:
this level of wrapping is confusing to me - it seems like if I got a `TableServiceError` all I'd ever want to do is grab the `odataError` property -- so why have this extra level of object? Why not just return a `TableServiceErrorOdataError` directly?

Reply:
Oh I see this is the generated code...

Reply:
Should we unwrap the error object in the convenience layer?

Reply:
Yes! It doesn't make sense to have this lone property, we should flatten it. I'll take care of this with #10378